### PR TITLE
Update clark client to reflect changes in core structure changes NUMBER 2 DRAFT FOR MIKE

### PR DESCRIPTION
**Pull request by stef359**

### CODE REVIEW

**Comment by stef359** [comment]
I wanted to let you know that the _**utility and notification modules**_ are the best examples of the new correct folder structure
and new fully constructed service. The rest are a bit messy or not started at the moment so they aren't the best examples for demonstration.

File: src/app/core/collection-module/collections.service.ts [hunk 1]
```diff
@@ -145,15 +146,6 @@ export class CollectionService {
...
-    return this.http.get(COLLECTIONS_ROUTES.GET_COLLECTION_METRICS(name))
-      .pipe(
-        retry(3),
-        catchError(this.handleError)
-      )
-      .toPromise();
-
-  }
   getCollectionCuratorsInfo(name: string) {
     return this.http.get(COLLECTIONS_ROUTES.GET_COLLECTION_CURATORS(name))
```
Comment (mfranz1) [inline src/app/core/collection-module/collections.service.ts:150]: This needs to use the new route

File: src/app/core/collection-module/collections.service.ts [hunk 2]
```diff
@@ -234,7 +226,7 @@ export class CollectionService {
       start: string,
       end: string
     }) {
-    if(collections.length > 0) {
+    if (collections.length > 0) {
       const route = COLLECTIONS_ROUTES.GET_COLLECTION_REPORT(collections, date);
```
Comment (mfranz1) [inline src/app/core/collection-module/collections.service.ts:230]: This needs to use the new route. This one is in the reports moduel

File: src/app/core/feature-module/featured.service.ts [hunk 3]
```diff
@@ -165,7 +166,7 @@ export class FeaturedObjectsService {
    */
   getCollectionFeatured(collection: string) {
     return this.http
-      .get(FEATURED_ROUTES.GET_COLLECTION_FEATURED(collection))
+      .get(FEATURED_ROUTES_OLD.GET_COLLECTION_FEATURED(collection))
```
Comment (mfranz1) [inline src/app/core/feature-module/featured.service.ts:169]: This should use the new route

File: src/app/core/standard-guidelines-module/guideline.service.ts [hunk 4]
```diff
@@ -15,7 +15,7 @@ export class GuidelineService {
   constructor(public http: HttpClient) {
     // Load guideline sources only once
     this.sources = this.http
-      .get<{total: number, results: FrameworkDocument[]}>(STANDARD_GUIDELINE_ROUTES.SEARCH_FRAMEWORKS({page: '1', limit: '20'}))
```
Comment (mfranz1) [inline src/app/core/standard-guidelines-module/guideline.service.ts:18]: I don't think these query params should have been dropped

File: src/app/core/standard-guidelines-module/guideline.service.ts [hunk 5]
```diff
@@ -35,20 +35,20 @@ export class GuidelineService {
...
-      .then((res: {total: number, results: any[]}) => {
+      .then((res: { total: number, results: any[] }) => {
         return res;
       });
   }
 
   async getFrameworks(query?: any): Promise<FrameworkDocument[]> {
     return this.http
-      .get<{total: number, results: FrameworkDocument[]}>(STANDARD_GUIDELINE_ROUTES.SEARCH_FRAMEWORKS(query))
+      .get<{ total: number, results: FrameworkDocument[] }>(STANDARD_GUIDELINES_ROUTES.GET_FRAMEWORK(query))
```
Comment (mfranz1) [inline src/app/core/standard-guidelines-module/guideline.service.ts:51]: I don't think this is implemented correctly

File: src/app/core/user-module/profiles.service.ts
Comment (mfranz1) [inline src/app/core/user-module/profiles.service.ts:1]: Was this rewritten?

File: src/app/core/user-module/user.service.ts [hunk 6]
```diff
@@ -110,36 +81,30 @@ export class UserService {
         USER_ROUTES.REMOVE_COLLECTION_MEMBER(collection, memberId),
```
Comment (mfranz1) [inline src/app/core/user-module/user.service.ts:81]: This needs to be updated

File: src/app/core/user-module/user.service.ts [hunk 7]
```diff
@@ -110,36 +81,30 @@ export class UserService {
...
    * Performs a text search and returns a list of matching users
    *
    * @param {string} query the text string to query by
-   * @returns {Promise<User[]} array of users matching the text query
+   * @returns {Promise<User[]>} array of users matching the text query
    * @memberof UserService
    */
   searchUsers(query: UserQuery): Promise<User[]> {
     return this.http
       .get(USER_ROUTES.SEARCH_USERS(query), {
```
Comment (mfranz1) [inline src/app/core/user-module/user.service.ts:100]: This needs to be updated

File: src/app/core/user-module/user.service.ts [hunk 8]
```diff
@@ -154,14 +119,11 @@ export class UserService {
     const route = USER_ROUTES.GET_SAME_ORGANIZATION(organization);
```
Comment (mfranz1) [inline src/app/core/user-module/user.service.ts:119]: This needs to be updated

File: src/app/core/utility-module/blogs-component.service.ts [hunk 9]
```diff
@@ -11,13 +11,13 @@ export class BlogsComponentService {
 
   recentBlog: Blog; // holds the latest blogpost
 
-  constructor(private blogsService: BlogsService) {
-    this.blogsService.getMostRecentBlog().subscribe(
```
Comment (mfranz1) [inline src/app/core/utility-module/blogs-component.service.ts:15]: Just confirming getMostREcentBlog and getAllBlogs are the same?

File: src/app/cube/core/usage-stats/usage-stats.service.ts [hunk 10]
```diff
@@ -28,7 +29,7 @@ export class UsageStatsService {
         )
         .toPromise(),
       this.http
-        .get<{metrics: any}>(STATS_ROUTES.LIBRARY_METRICS)
```
Comment (mfranz1) [inline src/app/cube/core/usage-stats/usage-stats.service.ts:31]: Just confirming these are the same routes?

File: src/app/cube/cube.module.ts [hunk 11]
```diff
@@ -9,7 +9,7 @@ import { CubeRoutingModule } from './cube.routing';
 
 // Services
 import { LearningObjectService } from './learning-object.service';
-import { CoreModule } from 'app/core/core.module';
+//import { CoreModule } from 'app/core/core.module';
```
Comment (mfranz1) [inline src/app/cube/cube.module.ts:12]: Same question did you notice any degraded performace

File: src/app/cube/user-profile/components/profile-header/profile-header.component.ts [hunk 12]
```diff
@@ -76,7 +73,7 @@ export class ProfileHeaderComponent implements OnInit {
    * Method to update user info from service after changes have been made
    */
   async updateInfo() {
-    this.user = await this.profileService.fetchUserProfile(this.user.username);
+    this.user = await this.userService.fetchUserProfile(this.user.username);
```
Comment (mfranz1) [inline src/app/cube/user-profile/components/profile-header/profile-header.component.ts:76]: I would double check these service functions were doing the same thing

Comment (stef359) [reply src/app/cube/user-profile/components/profile-header/profile-header.component.ts:76]: I only changed the location of the service function from profile services to user user services.

File: src/app/cube/core/usage-stats/usage-stats.service.ts (hunk 10 above)
Comment (stef359) [reply src/app/cube/core/usage-stats/usage-stats.service.ts:31]: Yes.
STATS_ROUTES.LIBRARY_METRICS = ${environment.apiURL}/learning-objects/metrics
METRIC_ROUTES.GET_LEARNING_OBJECT_METRICS() =  ${environment.apiURL}/learning-objects/metrics

File: src/app/core/user-module/user.service.ts (hunk 8 above)
Comment (stef359) [reply src/app/core/user-module/user.service.ts:119]: I just deleted the route, it's not used anywhere.

File: src/app/core/auth-module/auth.service.ts [hunk 13]
```diff
@@ -524,19 +484,13 @@ export class AuthService {
...
-        {
-          headers: this.httpHeaders,
-          withCredentials: true
-        }
-      )
-      .pipe(
-        retry(3),
-        catchError(this.handleError)
-      )
+      .get(environment.apiURL + '/users/identifiers/active?email=' + email, {
```
[the remark this answers is not in the document]
Comment (stef359) [reply src/app/core/auth-module/auth.service.ts:489]: 👍

File: src/app/core/utility-module/blogs-component.service.ts (hunk 9 above)
Comment (stef359) [reply src/app/core/utility-module/blogs-component.service.ts:15]: Left a comment, `CLARK-SERVICE needs to accept a body with a get route, which currently it does not`, 
so getting the most recent blogs is unavailable for now.

File: src/app/components/blogs/blogs.component.ts [hunk 14]
```diff
@@ -35,11 +35,11 @@ export class BlogsComponent implements OnInit {
   @Output() showBlogsBanner: EventEmitter<boolean> = new EventEmitter();
   @Output() neverShowBanner: EventEmitter<{val: boolean, recentBlog?: Blog}> = new EventEmitter();
 
-  constructor(private blogsService: BlogsService) { }
+  constructor(private utilityService: UtilityService) {}
 
   ngOnInit(): void {
-    this.blogObservable = this.blogsService.getMostRecentBlog();
-    this.blogsService.getMostRecentBlog().subscribe(
```
[the remark this answers is not in the document]
Comment (stef359) [reply src/app/components/blogs/blogs.component.ts:42]: Left a comment, `CLARK-SERVICE needs to accept a body with a get route, which currently it does not`,
so getting the most recent blogs is unavailable for now.

File: src/app/admin/pages/users/users.component.ts [hunk 15]
```diff
@@ -76,7 +78,7 @@ export class UsersComponent implements AfterViewInit {
    */
   async fetchReviewers() {
     this.loading = true;
-    this.users = await this.user.fetchReviewers(this.activeCollection.abvName);
+    this.users = await this.access.fetchReviewers(this.activeCollection.abvName);
```
[the remark this answers is not in the document]
Comment (stef359) [reply src/app/admin/pages/users/users.component.ts:81]: I changed placement of the `fetchReviewers` service function from the user.service.ts to access-group.service.ts

File: src/app/core/auth-module/auth.service.ts (hunk 13 above)
[the remark this answers is not in the document]
Comment (stef359) [reply src/app/core/auth-module/auth.service.ts:489]: Updated: I took out the route VALIDATE_IDENTIFIER route and added two new routes, VALIDATE_EMAIL and VALIDATE_USERNAME, them implemented them in their right places.

**Pull request merged**